### PR TITLE
Feature/delete rows

### PR DESCRIPTION
**Pull request by Notargets**

### CODE REVIEW

File: executor/writer.go [hunk 1]
```diff
@@ -100,16 +103,17 @@ func (w *Writer) WriteRecords(ts []time.Time, data []byte) {
 			if err := w.AddNewYearFile(year); err != nil {
 				panic(err)
 			}
+			wkp = ThisInstance.WALFile.FullPathToWALKey(w.tbi.Path)
```
Comment (Notargets) [inline executor/writer.go]: This commit actually improves nothing - the optimization was there already and I didn't see it - it's a tad bit cleaner though :-)

File: executor/delete.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		*/
+		bufferSize := int(fp.Length + int64(iop.RecordLen))
+		buffer := make([]byte, bufferSize)
+		n, err := f.Read(buffer)
+		if err != nil || n != bufferSize {
+			return fmt.Errorf("delete(): Short read %d bytes", n)
+		}
+		numRecs := bufferSize / int(iop.RecordLen)
+		zeroRecord := make([]byte, int(iop.RecordLen))
+		var isContiguous bool
```
Comment (umitanuki) [inline executor/delete.go]: initialize with `false` just in case?

**Comment by umitanuki** [comment]
@Notargets thanks for quick work. I'll test it but looks like the same as I was imagining. So we don't touch variable length data that's already written and just unlink index for now.

File: executor/writer.go (hunk 1 above)
Comment (umitanuki) [reply executor/writer.go]: Yeah who knows. We've seen a couple of memory leak crashes recently elsewhere due to small oversight. Keeping it tidy to minimize the memory use is always good.

**Pull request merged**